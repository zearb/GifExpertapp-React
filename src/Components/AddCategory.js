import React,{useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        // console.log(e.target.value);
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        // console.log(inputValue);
        if(inputValue.trim().length>1){
            
            setCategories( categ => [inputValue,...categ] );
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p style={{display:'none'}}> {inputValue} </p>
            <input type="text" value={inputValue} onChange={handleInputChange} />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired
} 