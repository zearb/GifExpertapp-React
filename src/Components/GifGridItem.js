import React from 'react';
import PropTypes from 'prop-types';

export default function GifGridItem({title,url}) {

    
    // console.log(props);

    return (
        <div className="card animate__animated animate__zoomIn">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}