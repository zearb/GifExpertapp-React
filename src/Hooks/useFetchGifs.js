import { useState, useEffect } from "react";
import {getGif} from '../Helpers/getGif';


export const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data:[],
        loading:true
    });

    useEffect( () => {

        getGif(category)
            .then( (gifs) => {

                    
                setState({
                    data:gifs,
                    loading:false
                });


            } );

    }, [category] );


    return state;

}