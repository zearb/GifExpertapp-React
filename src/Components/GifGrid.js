import React,{} from 'react';
import { useFetchGifs } from '../Hooks/useFetchGifs';
import PropTypes from 'prop-types';
import GifGridItem from './GifGridItem';

export const GifGrid = ({category}) => {

    const {data:images,loading} = useFetchGifs(category);
    // console.log(data);


    return (
        <div>
            <h3 className="animate__animated animate__zoomIn">{category}</h3>

            {loading && <p className="animate__animated animate__flash animate__infinite infinite animate__slow	2s">Cargando...</p>}

            <div className="cont_cards">
                {
                    images.map( img => 
                    
                        <GifGridItem
                            key = {img.id}
                            {...img}
                        />

                    )
                }
            </div>
        </div>
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}