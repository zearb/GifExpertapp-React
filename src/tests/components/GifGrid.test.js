import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { GifGrid } from '../../Components/GifGrid';
import { useFetchGifs } from '../../Hooks/useFetchGifs';

jest.mock('../../Hooks/useFetchGifs');

describe('Pruebas en componente <GifGrid/>', () => {
    
    const category = 'DBZ';

    test('Se debe mostrar correctamente',()=>{

        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });
        
        const wrapper = shallow(<GifGrid category = {category} />);

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar items cuando se cargan imÃ¡genes con useFetchGifs',()=>{

        const gifs = [{
             id: 'ABC',
             title: 'DBZ' ,
             url: 'https//:localhost/unaimagen.jpg'
        }]
        
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });

        const wrapper = shallow(<GifGrid category = {category} />);

        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    });


});
