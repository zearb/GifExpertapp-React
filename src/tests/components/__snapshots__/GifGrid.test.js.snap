// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifGrid/> Se debe mostrar correctamente 1`] = `
<div>
  <h3
    className="animate__animated animate__zoomIn"
  >
    DBZ
  </h3>
  <p
    className="animate__animated animate__flash animate__infinite infinite animate__slow	2s"
  >
    Cargando...
  </p>
  <div
    className="cont_cards"
  />
</div>
`;
